src/libsync/vfs/cfapi/shellext/thumbnailprovideripc.h

```diff
@@ -0,0 +1,39 @@
+/*
+ * Copyright (C) by Oleksandr Zolotov <alex@nextcloud.com>
+ *
+ * This program is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation; either version 2 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful, but
+ * WITHOUT ANY WARRANTY; without even the implied warranty of MERCHANTABILITY
+ * or FITNESS FOR A PARTICULAR PURPOSE. See the GNU General Public License
+ * for more details.
+ */
+
+#pragma once
+
+class QString;
+class QSize;
+class QLocalSocket;
+
+#include <QByteArray>
+
+namespace CfApiShellExtensions {
+class ThumbnailProviderIpc
```

Comment:
should we not have this directly in `libsync` ?
nothing looks specific to CfApi and probably could be useful on other platforms

Reply:
@mgallien Well, the thumbnail provider can work without Virtual Files, it is true. The shell extension, however, is specific to `SyncRootManager`, as we register it for each folder that is a Virtual Files Sync Root.

Do you know how the same feature works on other platforms? I don't have such information, therefore I don't see how this can be useful for Linux or mac for example (especially since we don't have Virtual Files implemented for mac). So, I'd say, it could be moved to a higher level as soon as we are sure the approach for thumbnails is the same for other platforms and if the concept of thumbnails does exist on those platforms. Or, did you mean something else?

Reply:
but it is very weird from a design point of view to have this PR changes classes unrelated to virtual files and at the same time have the implementation specific to virtual files
I would still suggest to keep everything not using the virtual files low level microsoft API without link to CfApi and only add in CfApi virtual files plugin what is really specific to the API

Reply:
I confused this part with the server implementation of the protocol
closing this thread